crates/task-impls/src/consensus.rs

```diff
@@ -609,11 +610,13 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                             // starting from the first iteration with a three chain, e.g. right after the else if case nested in the if case above
                             if new_decide_reached {
                                 let mut leaf = leaf.clone();
-                                consensus
-                                    .metrics
-                                    .last_synced_block_height
-                                    .set(usize::try_from(leaf.get_height()).unwrap_or(0));
-
+                                if !is_ancestor {
```

Comment:
You could also just check `if leaf.view_number == new_anchor_view {`?

Reply:
+1 I feel checking the view number is better than relying on the `is_ancester` flag.

Reply:
+1 I think this resolves my comment to, I mostly wanted to avoid the extra `is_ancester` flag as well

Reply:
Aha!